go/proof.go

```diff
@@ -66,6 +59,23 @@ func (p *ExistenceProof) Verify(spec *ProofSpec, root CommitmentRoot, key []byte
 
 }
 
+// VerifyItem checks that the proof is a valid proof of a (key, value) for this root
+func (p *ExistenceProof) VerifyItem(spec *ProofSpec, root CommitmentRoot, key, value []byte) error {
```

Comment:
VerifyItem then does both `Verify` and that the proof actually proves what we think it does.